### DISCUSSION TITLE
PWA apps and Workbox service workers conflict

### DISCUSSION THREAD

**Discussion by arekucr** [mswjs/msw#1280] [Q&A]
Does this library work if the site is a PWA that already use a service worker with workbox? I understand that a site cant have 2 service workers, but also this is for development, and the workbox only is generated after we build the site. I ask this bc were thinking in use this library so we can work in a feature no matter if we dont have the API side ready (this is made by other team) In that way QA can start testing in the first test enviroment using mock data and with the data layer to simulate the app is working normally. But the site running in the test env will be a PWA with its own SW with workbox,

**Comment by kettanaito** [comment]
Hi, @arekucr. 

To my best knowledge, there can be only one service worker controlling one host. If you're already using a service worker, for example from the tools like workbox, then you'd have to compose them into a single module using [`importScripts`](https://developer.mozilla.org/en-US/docs/Web/API/WorkerGlobalScope/importScripts). Take a look at the related discussion #1034 on how to do that. 

I believe it should be possible to include another worker within the generated PWA worker. I haven't tried this myself, so please let me know how it goes.
